portmap/core/admin.py

```diff
@@ -62,4 +68,54 @@ class CustomUserAdmin(UserAdmin):
 
 @register(TrackArticleView, site=admin_site)
 class TrackArticleViewAdmin(admin.ModelAdmin):
-    list_display = ("article", "article_path", "visited_directly", "external_referrer", )  
+    list_display = ("article", "article_path", "visited_directly", "external_referrer", )
+
+@staff_member_required
+def analytics(request):
+    datatype_interest_counts = {}
+    datatype_provider_query_counts = {}
+
+    for article in Article.objects.all():
+        if article.datatype not in datatype_interest_counts:
+            datatype_interest_counts[article.datatype] = {
+                'views': 0,
+                'queries': 0
+            }
+
+    for query in QueryLog.objects.all():
```

Comment:
ISTM our data model could be better for creating statistics like this rather than build counts?  That said, it's fine like this, just a note for the future.